module/s3-scan-object/app.js

```diff
@@ -110,7 +116,9 @@ exports.handler = async (event) => {
         tags.push({ Key: "av-checksum", Value: scanChecksum });
       }
 
-      isObjectTagged = await tagS3Object(s3Client, s3Object, tags);
+      roleArn = `arn:aws:iam::${awsAccountId}:role/${AWS_ROLE_TO_ASSUME}`;
+      s3Clients[awsAccountId] = await getS3Client(s3Clients[awsAccountId], stsClient, roleArn);
```

Comment:
any chance we'll run into a scenario where we have cached credentials and the lambda stays warm long enough for the credentials to expire ?

If so we might want to enhance the getS3Client to test the creds are valid when a cached result is found or store timeout/expirty lists.

Reply:
I think we'll be ok because of how the cache is only created within the handler, so should only be valid for the single invocation.

In all honesty, this is probably overkill since I think almost all our lambda invoke will be with payloads for a single AWS account ID...but, I do love me some edge cases.